mysql/mysql_probe.go

```diff
@@ -37,10 +37,10 @@ func (i MysqlProbe) DoProbe(instance *dockerinitiator.Instance) error {
 	}()
 
 	db, err := sql.Open("mysql", fmt.Sprintf("%s:%s@tcp(%s)/%s", "root", i.Password, instance.GetHost(), i.DbName))
-	defer db.Close()
 	if err != nil {
 		return err
 	}
+	defer db.Close()
```

Comment:
I've had this error before 😅 
Can be a bit annoying and hard to find.

I did a git blame and saw that it was me who put it there... 🙃

Reply:
Hehe, simple mistake to make! Luckily the Open hasn't errored that often yet!